app/src/main/java/com/soen490chrysalis/papilio/view/UserProfileActivity.kt

```diff
@@ -71,6 +129,270 @@ class UserProfileActivity : AppCompatActivity()
+            passwordInput2.hint = "Enter New Password"
+
+            val passwordInput3 = EditText(this)
+            passwordInput3.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+            passwordInput3.hint = "Confirm New Password"
+
+            val messageText = TextView(this)
+            messageText.setTextColor(Color.RED)
+
+            val layout = LinearLayout(this)
+
+            val params = LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.MATCH_PARENT
+            )
+
+            layout.orientation = LinearLayout.VERTICAL
+            layout.setPadding(50, 75, 50, 75)
+            layout.layoutParams = params
+
+            layout.addView(passwordInput1)
+            layout.addView(passwordInput2)
+            layout.addView(passwordInput3)
+            layout.addView(messageText)
+
+
+            builder.setView(layout)
+
+            builder.setPositiveButton(
+                "Save",
+                DialogInterface.OnClickListener { dialog_ : DialogInterface, _ : Int ->
+
+                })
+
+            builder.setNegativeButton(
+                "Cancel",
+                DialogInterface.OnClickListener { dialog : DialogInterface, _ : Int ->
+                    run {
+                        // When the "Accept button is pressed, simply dismiss the dialog
+                        dialog.dismiss()
+                    }
+                })
+
+            val dialog = builder.create()
+
+            dialog.show()
+
+            dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener(View.OnClickListener {
+
+                var closeDialog = false
+
+
+                var validationError : String? = null
+                var confirmationError : String? = null
+
+                validationError = userProfileViewModel.validatePassword(passwordInput2.text.toString())
+                confirmationError = userProfileViewModel.confirmPassword(passwordInput2.text.toString(), passwordInput3.text.toString())
+
+                if(validationError != null)
+                {
+                    messageText.setText(validationError)
+                }
+                else if (confirmationError != null)
+                {
+                    messageText.setText(confirmationError)
+                }
+                else if(passwordInput1.text == null)
+                {
+                    messageText.setText("Please enter current password")
+                }
+                else
+                {
+                    userProfileViewModel.changeUserPassword(passwordInput1.text.toString(), passwordInput3.text.toString())
+                    closeDialog = true
+                }
+
+                if (closeDialog) dialog.dismiss()
+            })
+
+        }
+
+        binding.userProfileAddPhoneButton.setOnClickListener{
+            val builder = AlertDialog.Builder(this)
```

Comment:
You are repeating this entire dialog creation code, maybe you can create a function that takes parameters such as title 
and a list of input fields to reuse code